ssas/systems.go

```diff
@@ -537,6 +537,20 @@ func GenerateSecret() (string, error) {
 	return fmt.Sprintf("%x", b), nil
 }
 
+func GetAllIPs() ([]string, error) {
+	var (
+		db      = GetGORMDbConnection()
+		ips 	[]string
+		err     error
+	)
+	defer Close(db)
+
+	if err = db.Order("address").Model(&IP{}).Where("deleted_at IS NULL AND id NOT IN (SELECT id FROM systems WHERE deleted_at IS NOT NULL)").Pluck("DISTINCT address", &ips).Error; err != nil {
```

Comment:
Interesting.  Wasn't expecting an inner `select` on the `systems` table, but I think it makes a lot of sense.  Should this be commented appropriately?

Reply:
Comment added to latest commit